pkg/vsphere/session/session.go

```diff
@@ -282,6 +326,16 @@ func (s *Session) Populate(ctx context.Context) (*Session, error) {
 		log.Debugf("Cached pool: %s", s.PoolPath)
 	}
 
+	if s.Datacenter != nil {
+		folders, err := s.Datacenter.Folders(ctx)
+		if err != nil {
+			errs = append(errs, fmt.Sprintf("Failure finding folders (%s): %s", s.DatacenterPath, err.Error()))
+		} else {
+			log.Debugf("Cached folders: %s", s.DatacenterPath)
+		}
+		s.VMFolder = folders.VmFolder
```

Comment:
This is being used without being checked elsewhere - should be a hard error if it cannot be set.